f/mapClickTeleport/fn_mapClickTeleportParachute.sqf

```diff
@@ -2,22 +2,70 @@
 // Credits: Please see the F3 online manual (http://www.ferstaberinde.com/f3/en/)
 // ====================================================================================
 
-//add aparachute as backpack and restore the old backpack on landing
-
-private ["_bp","_bpi"];
-params [
-	["_unit", objNull, [objNull]]
-];
-_bp = backpack _unit;
-_bpi = backPackItems _unit;
-
-removeBackpack _unit;
-_unit addBackpack "B_parachute";
-waitUntil {sleep 0.1;isTouchingGround _unit;};
-if (alive _unit) then {
-	removeBackpack _unit;
-	_unit addBackPack _bp;
-	{
-		(unitbackpack _unit) addItemCargoGlobal [_x,1];
-	} forEach _bpi;
+// If a vehicle is teleported, a parachute is attached at a given height.
+// If it's a player, a parachute is added as a backacpk.
+// After landing, the old backpack is restored.
+
+params [["_obj", objNull, [objNull]]];
+
+private _heightOpenParachute = 200;
+
+if ( ! (_obj isKindOf "CAManBase") ) then {
+
+	// Set the vehicle up-right to make sure it doesn't drift away too much
+	_obj setVectorUp [0, 0, 1];
+
+	waitUntil { sleep 0.1; getPos _obj select 2 < _heightOpenParachute };
+	_obj engineOn false;
+	
+	//Set the vehicle captive to prevent AT units from shooting it down immediately
+	_obj setCaptive true;
+
+	// Create the parachute, and attach the vehicle.
+	private _chute = createVehicle ["B_Parachute_02_F", getPos _obj, [], 0, "NONE"];
+	_chute setDir getDir _obj;
+	_obj attachTo [_chute, [0, 0, 0.2]];
+
+	// Once near the ground, detach the parachute  and gently place the vehicle on the ground
+	waitUntil { sleep 0.1; getPos _obj select 2 < 1.5 || {isNull _chute || {! alive _chute}} };
+	_obj allowDamage false;
+	detach _obj;
+	_obj setVectorUp [0, 0, 1];
+	_obj setVelocity [0, 0, -1];
+	_obj setPos [getPos _obj select 0, getPos _obj select 1, 0.5];
+
+	// Let the parachute fall down
+	_chute setVelocity [0 - sin windDir * 5, 0 - cos windDir * 5, 0];
+	_chute disableCollisionWith _obj;
+
+	//After having landed, disable captivity, re-enable damage and remove the parachute
+	sleep 10;
+	_obj setCaptive false;
+	_obj allowDamage true;
+
+	if (!isNull _chute) then {
+		deleteVehicle _chute;
+	};
+
+} else {
+
+	// Save the backpack and its items
+	private _backpack = backpack _obj;
+	private _backpackItems = backPackItems _obj;
+
+	// Replace the backpack with a parachute
+	removeBackpack _obj;
+	_obj addBackpack "B_parachute";
+
+	// Once on the ground, remove the parachute 
+	// and re-add the backpack with all its content
+	waitUntil {sleep 0.1; isTouchingGround _obj};
```

Comment:
`isTouchingGround` can be a bit temperamental. We'll see how it goes in testing but might be good to have a backup such as detecting when the player drops their parachute.

Also, do you keep the parachute backpack after landing?

Reply:
we have always used `isTouchingGround` if you look at the diff. This isn't something new.

Reply:
`Also, do you keep the parachute backpack after landing?`
A few lines below, the parachute is replaced with the original backpack. like it has always been.

Reply:
I meant, what is the vanilla behaviour?

Reply:
`I do believe they keep it on their back after landing.`
Cool, was just checking. If `isTouchingGround` has always been used, no need to change it.